### DISCUSSION TITLE
References to time servers

### DISCUSSION THREAD

**Discussion by NitramH** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#336] [General]
Hi all,

my ESP is running behind the firewall, with no access to the internet (in my opinion not needed).
If I want to access from outside I do this using a VPN connection to my router.

The problem is, that the software connects at several points to time servers (maybe also in the libraries?)
I would like to point to my router, which is acting as own time server.
Wihtout the correct time the filter and chlorine timer reacts strange.

Does someone knows where in the code time server references are located?
I tried to search but it is hard to find (time is a very often used word :-( )

Martin

**Comment by chunkysteveo** [comment]
main.cpp file there is a call to "pool.ntp.org". (line 384-ish for 4-wire, line 399-ish 6-wire). 

I am guessing you can call a custom NTP time server URL there if you have your own?

**Reply by chunkysteveo** [reply]
Actually - update the URL in BWC_common.cpp file, about line 543 (6-wire), 622 (4-wire). 

```DateTime.setServer("pool.ntp.org");```

You should be able to update it there and it will take the custom URL?

**Reply by NitramH** [reply]
changing it in main.cpp was enough.
Seems that the setting in BWC_common.cpp is overwritten later by main.cpp

Yes, I entered the IP of my router and it has the correct time now.
Thanks a lot.
